recipe/meta.yaml

```diff
@@ -32,6 +47,19 @@ requirements:
     - libgrpc {{ libgrpc }}
 
   host:
+    - cudnn             # [cuda_compiler_version != "None"]
+    - nccl              # [cuda_compiler_version != "None"]
+    - cuda-version {{ cuda_compiler_version }}  # [cuda_compiler_version != "None"]
+    - cuda-cupti-dev    # [(cuda_compiler_version or "").startswith("12")]
+    - cuda-cudart-dev   # [(cuda_compiler_version or "").startswith("12")]
+    - cuda-nvml-dev     # [(cuda_compiler_version or "").startswith("12")]
+    - cuda-nvtx-dev     # [(cuda_compiler_version or "").startswith("12")]
+    - cuda-nvcc-tools  # [(cuda_compiler_version or "").startswith("12")]
+    - libcublas-dev     # [(cuda_compiler_version or "").startswith("12")]
+    - libcusolver-dev   # [(cuda_compiler_version or "").startswith("12")]
+    - libcurand-dev     # [(cuda_compiler_version or "").startswith("12")]
+    - libcufft-dev      # [(cuda_compiler_version or "").startswith("12")]
+    - libcusparse-dev   # [(cuda_compiler_version or "").startswith("12")]
```

Comment:
I can't tell if this is clever or likely to cause problems when we build with a compiler other than 12.x.

Reply:
I don't love it, but I stuck with it to match our other feedstocks (ie tensorflow) and CF.

Reply:
yeah this is an ok pattern for conda-forge because the cuda_compiler_version (along with c_compiler_version) is set locally in the feedstock, via auto rendered config files. [example](https://github.com/conda-forge/lightgbm-feedstock/blob/5e71ffaf89e9585c69eda259deca369434e4f67f/.ci_support/linux_64_c_compiler_version12cuda_compilercuda-nvcccuda_compiler_version12.0cxx_compiler_version12.yaml#L17). We define c_compiler_version in our aggregate conda_build_config.yaml and I think should probably do the same with the cuda compiler. (we currently define `cuda_compiler` there). So I don't think this pattern works as well for us.